tests/request_size_test.py

```diff
@@ -0,0 +1,167 @@
+    database_dir = operation_dir / 'database'
+    database_dir.mkdir(exist_ok=True, parents=True)
+    tests.database.initialize_environment(database_dir=database_dir)
+
+    # create and store a mock API key
+    api_key = uuid.uuid4()
+    api_key_pth = operation_dir / 'key'
+    with api_key_pth.open(mode='w') as fid:
+        fid.write(str(api_key))
+
+    port_ctl = tests.component_test.find_free_port()
+
+    # start the control server to store a channel in the empty database
+    # yapf: disable
+    cmd = [str(release_dir / 'bin' / 'mailgun-relay-controlery'),
+           '-database_dir', database_dir.as_posix(),
+           '-address', ':{}'.format(port_ctl)]
+    # yapf: enable
+    if quiet:
+        cmd += '-quiet'
+
+    desc = "some-channel-name"
+    token = "leWq221234123423oiweoWPEOFIWPEOFKPDKlsdkepwsodPOR"
+
+    stdout = subprocess.DEVNULL if quiet else None
+
+    proc = subprocess.Popen(cmd, stdout=stdout)
+    with tests.proc.terminating(proc=proc, timeout=5):
+        # let the server initialize
+        tests.proc.sleep_while_process(proc=proc, seconds=1)
+
+        if proc.poll() is not None:
+            raise AssertionError("Expected the server process to be alive, but it died.")
+
+        url = 'http://127.0.0.1:{}'.format(port_ctl)
+        client = tests.control.RemoteCaller(url)
+
+        # add a channel
+        sender = tests.control.Entity(email="someone@some-domain.com")
+        recipients = [
+            tests.control.Entity(name="client", email="client@another-domain.com"),
+        ]
+        cc_field = [tests.control.Entity(name="client-2", email="client-2@another-domain.com")]
+        bcc_field = [tests.control.Entity(name="devop-2", email="devop-2@some-domain.com")]
+        channel = tests.control.Channel(
+            descriptor=desc,
+            token=token,
+            sender=sender,
+            recipients=recipients,
+            cc=cc_field,
+            bcc=bcc_field,
+            domain="requests.test.com",
+            min_period=0.001,
+            max_size=100000000)
+        client.put_channel(channel=channel)
+
+    # start the relayery server
+    port_relay = tests.component_test.find_free_port()
+
+    # yapf: disable
+    cmd = [str(release_dir / 'bin' / 'mailgun-relayery'),
+           '-database_dir', database_dir.as_posix(),
+           '-api_key_path', api_key_pth.as_posix(),
+           '-address', ':{}'.format(port_relay)]
+    # yapf: enable
+    if quiet:
+        cmd += '-quiet'
+
+    proc = subprocess.Popen(cmd, stdout=stdout)
+    with tests.proc.terminating(proc=proc, timeout=5):
+        # let the server initialize
+        tests.proc.sleep_while_process(proc=proc, seconds=1)
+
+        if proc.poll() is not None:
+            raise AssertionError("Expected the server process to be alive, but it died.")
+
+        # relay a message with wrong auth token and large attachment (size: 1 Gb)
+        with tempfile.NamedTemporaryFile() as fid:
```

Comment:
You actually don't have to create the file. You only need to figure out which functions `MultipartEncoder` will actually use.

Then you can just create a mock class with these methods that returns a couple of zero bytes and sleeps for ten seconds before sending the other chunk of a couple of zero bytes.

You can then test that the time elapse for `post()` was less than, say, 2 seconds.